src/xcode/ENA/ENA/Source/Services/Risk/Provider/Model/RiskProvidingConfiguration.swift

```diff
@@ -41,12 +41,25 @@ extension RiskProvidingConfiguration {
 	}
 
 	func nextExposureDetectionDate(lastExposureDetectionDate: Date?, currentDate: Date = Date()) -> Date {
-		let potentialDate = Calendar.current.date(
+		// The lastExposureDetection should have been computed before:
+		guard let lastExposureDetectionDate = lastExposureDetectionDate else {
+			return Calendar.current.date(
+				byAdding: exposureDetectionInterval,
+				to: .distantPast,
+				wrappingComponents: false
+			) ?? .distantPast
+		}
+
+		// The lastExposureDetection should not be in the future.
+		guard lastExposureDetectionDate < currentDate else {
```

Comment:
In which case can this happen?

Reply:
i rewrote this code and one of the test made sure this is the behavior. I guess some other code calls this method with Date.distantPast when no other date is available.